Duplicate symbols when linking to static openal-soft

Linking re3 fails because duplicate openal symbols are detected.
These duplicate symbols are defined in `src/audio/oal/oal_utils.cpp` and in the (static) openal-soft library.
Won't this also cause undefined behavior with a shared openal-soft library?

https://github.com/GTAmodding/re3/blob/9e45feb4fa0a841b1d059bcab8507fd80708d6e7/src/audio/oal/oal_utils.cpp#L6-L38

Very related, `AUDIO_OAL` is always enabled in the cmake build.
https://github.com/GTAmodding/re3/blob/9e45feb4fa0a841b1d059bcab8507fd80708d6e7/src/CMakeLists.txt#L67
Is this correct? Shouldn't it only be enabled when the audio backend is openal (`RE3_AUDIO` == `OAL`)?
https://github.com/GTAmodding/re3/blob/9e45feb4fa0a841b1d059bcab8507fd80708d6e7/CMakeLists.txt#L12

<details>

<summary>Complete linking error log</summary>

```
[100%] Linking CXX executable ../../bin/re3
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alIsAuxiliaryEffectSlot':
auxeffectslot.cpp:(.text+0xa90): multiple definition of `alIsAuxiliaryEffectSlot'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x98): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alGetAuxiliaryEffectSloti':
auxeffectslot.cpp:(.text+0xbc0): multiple definition of `alGetAuxiliaryEffectSloti'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x70): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alGetAuxiliaryEffectSlotiv':
auxeffectslot.cpp:(.text+0xd70): multiple definition of `alGetAuxiliaryEffectSlotiv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x68): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alGetAuxiliaryEffectSlotf':
auxeffectslot.cpp:(.text+0xed0): multiple definition of `alGetAuxiliaryEffectSlotf'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x60): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alGetAuxiliaryEffectSlotfv':
auxeffectslot.cpp:(.text+0x1050): multiple definition of `alGetAuxiliaryEffectSlotfv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x58): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alAuxiliaryEffectSloti':
auxeffectslot.cpp:(.text+0x1190): multiple definition of `alAuxiliaryEffectSloti'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x90): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alAuxiliaryEffectSlotiv':
auxeffectslot.cpp:(.text+0x15b0): multiple definition of `alAuxiliaryEffectSlotiv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x88): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alAuxiliaryEffectSlotf':
auxeffectslot.cpp:(.text+0x1720): multiple definition of `alAuxiliaryEffectSlotf'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x80): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alAuxiliaryEffectSlotfv':
auxeffectslot.cpp:(.text+0x1910): multiple definition of `alAuxiliaryEffectSlotfv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x78): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alDeleteAuxiliaryEffectSlots':
auxeffectslot.cpp:(.text+0x1a60): multiple definition of `alDeleteAuxiliaryEffectSlots'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xa0): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(auxeffectslot.cpp.o): in function `alGenAuxiliaryEffectSlots':
auxeffectslot.cpp:(.text+0x2290): multiple definition of `alGenAuxiliaryEffectSlots'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xa8): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alDeleteEffects':
effect.cpp:(.text+0x5c0): multiple definition of `alDeleteEffects'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xf8): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alIsEffect':
effect.cpp:(.text+0xa60): multiple definition of `alIsEffect'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xf0): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alEffecti':
effect.cpp:(.text+0xb70): multiple definition of `alEffecti'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xe8): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alEffectiv':
effect.cpp:(.text+0xd50): multiple definition of `alEffectiv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xe0): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alEffectf':
effect.cpp:(.text+0xec0): multiple definition of `alEffectf'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xd8): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alEffectfv':
effect.cpp:(.text+0x1000): multiple definition of `alEffectfv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xd0): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alGetEffecti':
effect.cpp:(.text+0x1140): multiple definition of `alGetEffecti'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xc8): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alGetEffectiv':
effect.cpp:(.text+0x1290): multiple definition of `alGetEffectiv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xc0): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alGetEffectf':
effect.cpp:(.text+0x1400): multiple definition of `alGetEffectf'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xb8): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alGetEffectfv':
effect.cpp:(.text+0x1540): multiple definition of `alGetEffectfv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0xb0): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(effect.cpp.o): in function `alGenEffects':
effect.cpp:(.text+0x1680): multiple definition of `alGenEffects'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x100): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alDeleteFilters':
filter.cpp:(.text+0x870): multiple definition of `alDeleteFilters'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x48): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alIsFilter':
filter.cpp:(.text+0xd50): multiple definition of `alIsFilter'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x40): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alFilteri':
filter.cpp:(.text+0xe70): multiple definition of `alFilteri'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x38): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alFilteriv':
filter.cpp:(.text+0x1080): multiple definition of `alFilteriv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x30): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alFilterf':
filter.cpp:(.text+0x11f0): multiple definition of `alFilterf'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x28): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alFilterfv':
filter.cpp:(.text+0x1330): multiple definition of `alFilterfv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x20): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alGetFilteri':
filter.cpp:(.text+0x1480): multiple definition of `alGetFilteri'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x18): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alGetFilteriv':
filter.cpp:(.text+0x15e0): multiple definition of `alGetFilteriv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x10): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alGetFilterf':
filter.cpp:(.text+0x1750): multiple definition of `alGetFilterf'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x8): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alGetFilterfv':
filter.cpp:(.text+0x18a0): multiple definition of `alGetFilterfv'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x0): first defined here
/usr/bin/ld: /home/maarten/.conan/data/openal/1.20.1/_/_/package/8f758c7d79f11e0791dfc187e19cd92dffabbd2b/lib/libopenal.a(filter.cpp.o): in function `alGenFilters':
filter.cpp:(.text+0x19f0): multiple definition of `alGenFilters'; CMakeFiles/re3.dir/audio/oal/oal_utils.cpp.o:(.bss+0x50): first defined here
collect2: error: ld returned 1 exit status
```

</details>

--- a/src/audio/oal/oal_utils.cpp
+++ b/src/audio/oal/oal_utils.cpp
@@ -3,6 +3,14 @@
 
 #ifdef AUDIO_OAL
 
+/*
+ * When linking to a static openal-soft library,
+ * the extension function inside the openal library conflict with the variables here.
+ * Therefore declare these re3 owned symbols in a private namespace.
+ */
+
+namespace re3_openal {
+
 LPALGENEFFECTS alGenEffects;
 LPALDELETEEFFECTS alDeleteEffects;
 LPALISEFFECT alIsEffect;
@@ -37,6 +45,9 @@ LPALGETFILTERIV alGetFilteriv;
 LPALGETFILTERF alGetFilterf;
 LPALGETFILTERFV alGetFilterfv;
 
+}
+
+using namespace re3_openal;
 
 void EFXInit()
 {

--- a/src/audio/oal/oal_utils.h
+++ b/src/audio/oal/oal_utils.h
@@ -11,6 +11,8 @@ void EFX_Set(ALuint effect, const EAXLISTENERPROPERTIES *props);
 void EAX3_SetReverbMix(ALuint filter, float mix);
 void SetEffectsLevel(ALuint uiFilter, float level);
 
+namespace re3_openal {
+
 extern LPALGENEFFECTS alGenEffects;
 extern LPALDELETEEFFECTS alDeleteEffects;
 extern LPALISEFFECT alIsEffect;
@@ -44,4 +46,9 @@ extern LPALGETFILTERI alGetFilteri;
 extern LPALGETFILTERIV alGetFilteriv;
 extern LPALGETFILTERF alGetFilterf;
 extern LPALGETFILTERFV alGetFilterfv;
+
+}
+
+using namespace re3_openal;
+
 #endif

